.github/workflows/test-lint.yml

```diff
@@ -19,6 +19,23 @@ concurrency:
   cancel-in-progress: ${{ github.ref != 'refs/heads/main' }}
 
 jobs:
+  enforce-changeset:
+    name: Enforce Changeset
+    runs-on: ubuntu-latest
+    if: github.event_name == 'pull_request'
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+          ref: main
+      - run: git checkout ${{ github.event.pull_request.head.ref }}
+      - run: git branch
+      - name: install pnpm@8.3.1
+        run: npm i -g pnpm@8.3.1
+      - run: pnpm install
```

Comment:
Hopefully the two `pnpm install` will use the same filesystem and thus the second one will be cached.

If not, we might want to move this into the job below since you can use `if` conditions on `run`

Reply:
Hmm, seems like its not cached. Are they run concurrently?

https://github.com/vercel/vercel/actions/runs/5181960729/jobs/9338166740?pr=10065#step:5:12

https://github.com/vercel/vercel/actions/runs/5181960729/jobs/9338166887?pr=10065#step:6:12

Reply:
Ya they're concurrent. And would be different filesystems. I don't think it matters much to optimize here since other workflows will take longer anyways.